manager/orchestrator/constraint_enforcer.go

```diff
@@ -0,0 +1,137 @@
+package orchestrator
+
+import (
+	"github.com/docker/go-events"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/log"
+	"github.com/docker/swarmkit/manager/constraint"
+	"github.com/docker/swarmkit/manager/state"
+	"github.com/docker/swarmkit/manager/state/store"
+)
+
+// ConstraintEnforcer watches for updates to nodes and shuts down tasks that no
+// longer satisfy scheduling constraints or resource limits.
+type ConstraintEnforcer struct {
+	store       *store.MemoryStore
+	watcher     chan events.Event
+	cancelWatch func()
+	stopChan    chan struct{}
+	doneChan    chan struct{}
+}
+
+// NewConstraintEnforcer creates a new ConstraintEnforcer.
+func NewConstraintEnforcer(store *store.MemoryStore) *ConstraintEnforcer {
+	watcher, cancel := state.Watch(store.WatchQueue(), state.EventUpdateNode{})
+
+	return &ConstraintEnforcer{
+		store:       store,
+		watcher:     watcher,
+		cancelWatch: cancel,
+		stopChan:    make(chan struct{}),
+		doneChan:    make(chan struct{}),
+	}
+}
+
+// Run is the ConstraintEnforcer's main loop.
+func (ce *ConstraintEnforcer) Run() {
+	defer close(ce.doneChan)
+
+	for {
+		select {
+		case event := <-ce.watcher:
+			node := event.(state.EventUpdateNode).Node
+
+			var (
+				tasks []*api.Task
+				err   error
+			)
+
+			ce.store.View(func(tx store.ReadTx) {
+				tasks, err = store.FindTasks(tx, store.ByNodeID(node.ID))
+			})
+
+			if err != nil {
+				log.L.WithError(err).Errorf("failed to list tasks for node ID %s", node.ID)
+			}
+
```

Comment:
Yes my concern is the interleaving of actions might add unnecessary complexity in debug. We can't really tell which component is taking action. I know I comes back to ownership issue :(

Also if a node is set to `pause`, user expects no change on node so they can debug. But here ConstraintEnforcer actually starts reconciliation.